tapfreighter/wallet.go

```diff
@@ -1654,61 +1654,31 @@ func addAnchorPsbtInputs(btcPkt *psbt.Packet, vPkt *tappsbt.VPacket,
 	}
 
 	// Now that we've added an extra input, we'll want to re-calculate the
-	// total weight of the transaction, so we can ensure we're paying
-	// enough in fees.
-	var (
-		weightEstimator     input.TxWeightEstimator
-		inputAmt, outputAmt int64
-	)
-	for _, pIn := range btcPkt.Inputs {
-		inputAmt += pIn.WitnessUtxo.Value
-
-		inputPkScript := pIn.WitnessUtxo.PkScript
-		switch {
-		case txscript.IsPayToWitnessPubKeyHash(inputPkScript):
-			weightEstimator.AddP2WKHInput()
-
-		case txscript.IsPayToScriptHash(inputPkScript):
-			weightEstimator.AddNestedP2WKHInput()
-
-		case txscript.IsPayToTaproot(inputPkScript):
-			weightEstimator.AddTaprootKeySpendInput(
-				txscript.SigHashDefault,
-			)
-		default:
-			return fmt.Errorf("unknown pkScript: %x",
-				inputPkScript)
+	// total vsize of the transaction and the necessary fee at the desired
+	// fee rate.
+	inputScripts := make([][]byte, 0, len(btcPkt.Inputs))
+	for inputIdx, input := range btcPkt.Inputs {
+		// TODO(jhb): Support non-witness UTXOs as inputs by fetching
```

Comment:
I don't think we need this TODO here as I don't think we'll ever go through the effort of supporting non-witness inputs in our products. Or do you mean the `input.NonWitnessUtxo` field that might also be set?

Reply:
IIUC a non-witness input would have `NonWitnessUtxo` set, and `WitnessUtxo` unset, so this method for gathering the scripts would fail. If we don't want to support non-witness inputs (makes sense to me), then I agree the TODO can be removed. AFAICT it shouldn't be a problem if both are set.